### PR TITLE
Support Apache ShardingSphere-Elasticjob

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#5152 

___
### New feature or improvement
- I will finish at elastic-job 3.0.0-alpha publish

### CODE REVIEW

**Comment by wu-sheng** [comment]
What is the status of this? Any block?

**Comment by zhaoyuguang** [comment]
> What is the status of this? Any block?

Just wait apache elastic-job publish

**Comment by wu-sheng** [comment]
> Just wait apache elastic-job publish

FYI @tristaZero, when you have a release, please let us know. This depends on that.

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 1]
```diff
@@ -81,6 +81,7 @@
 * [Hystrix: Latency and Fault Tolerance for Distributed Systems](https://github.com/Netflix/Hystrix) 1.4.20 -> 1.5.12
 * Scheduler
   * [Elastic Job](https://github.com/elasticjob/elastic-job) 2.x
+  * [Apache Elastic Job](https://github.com/apache/shardingsphere-elasticjob) 3.0.0-alpha
```
Comment (tristaZero) [inline docs/en/setup/service-agent/java-agent/Supported-list.md:84]: Sorry, `Apache Elastic Job`? Maybe `Apache ShardingSphere-Elasticjob` is better?

**Comment by tristaZero** [comment]
@zhaoyuguang @wu-sheng Glad to see this PR. The latest version is expected to be released this week. I will give the update once it releases. Besides, welcome @zhaoyuguang to be my backup in case I forget. :)

File: docs/en/setup/service-agent/java-agent/Supported-list.md (hunk 1 above)
Comment (zhaoyuguang) [reply docs/en/setup/service-agent/java-agent/Supported-list.md]: had update, please recheck

**Comment by zhaoyuguang** [comment]
> @zhaoyuguang @wu-sheng Glad to see this PR. The latest version is expected to be released this week. I will give the update once it releases. Besides, welcome @zhaoyuguang to be my backup in case I forget. :)

Glad, And hope some day can become SS committer by contribution 👨‍💻‍

**Comment by tristaZero** [comment]
> > @zhaoyuguang @wu-sheng Glad to see this PR. The latest version is expected to be released this week. I will give the update once it releases. Besides, welcome @zhaoyuguang to be my backup in case I forget. :)
> 
> Glad, And hope some day can become SS committer by contribution 👨‍💻‍

Aha, looking forward to that as well!

**Comment by zhaoyuguang** [comment]
@tristaZero Friendly Ping, PR is ready.

**Comment by tristaZero** [comment]
> @tristaZero Friendly Ping, PR is ready.

Hi  @zhaoyuguang  Thanks for your reminder.

Apache ShardingSphere ElasticJob-3.0.0-alpha has been released successfully [1].
@wu-sheng It is the time to considering merging this PR, thanks.

[1] https://mvnrepository.com/artifact/org.apache.shardingsphere.elasticjob/elasticjob-lite

File: test/plugin/scenarios/elasticjob-3.x-scenario/config/expectedData.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      componentId: 24
+      isError: false
+      spanType: Entry
+      peer: ''
+      skipAnalysis: false
+      tags:
+        - {key: item, value: '0'}
+        - {key: shardingTotalCount, value: '1'}
+        - {key: taskId, value: not null}
+        - {key: shardingItemParameters, value: '{0=Beijing}'}
```
Comment (wu-sheng) [inline test/plugin/scenarios/elasticjob-3.x-scenario/config/expectedData.yaml:53]: @tristan-tsl Could you check whether these are recommended and important messages for the elastic job users? Any thing more?

File: apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                             MethodInterceptResult result) throws Throwable {
+        ShardingContexts shardingContexts = (ShardingContexts) allArguments[0];
+        Integer item = (Integer) allArguments[1];
+        String operateName = ComponentsDefine.ELASTIC_JOB.getName() + "/" + shardingContexts.getJobName();
+        AbstractSpan span = ContextManager.createEntrySpan(operateName, new ContextCarrier());
+        span.setComponent(ComponentsDefine.ELASTIC_JOB);
+        span.tag("item", item == null ? "" : String.valueOf(item));
+        span.tag("shardingTotalCount", Integer.toString(shardingContexts.getShardingTotalCount()));
+        span.tag("taskId", shardingContexts.getTaskId());
+        span.tag("shardingItemParameters", shardingContexts.getShardingItemParameters() == null ? "" : shardingContexts.getShardingItemParameters().toString());
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java:44]: @zhaoyuguang Tag key should be added as a constant in the plugin.

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java:44]: @tristaZero This is how we get these tags.

Comment (tristaZero) [reply apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java:44]: Hi @wu-sheng It seems these tags come from [ShardingContexts](https://github.com/apache/shardingsphere-elasticjob/blob/master/elasticjob-api/src/main/java/org/apache/shardingsphere/elasticjob/api/listener/ShardingContexts.java), which is a necessary job context for users. I am not familiar with what content skyWalking needs for its plugin, but if it wants to get core job information, I think it is enough. :)

**Comment by wu-sheng** [comment]
> Hi @wu-sheng It seems these tags come from ShardingContexts, which is a necessary job context for users. I am not familiar with what content skyWalking needs for its plugin, but if it wants to get core job information, I think it is enough. :)

We don't need anything. I just want to be sure, these are enough information for the ElasticJob users.

File: apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java (hunk 3 above)
Comment (tristaZero) [reply apm-sniffer/apm-sdk-plugin/elastic-job-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticjob/ElasticJobExecutorInterceptor.java:44]: BTW, there are other props from `ShardingContexts`, like `jobName`, `jobParameter`, and so forth. I am not sure whether you @zhaoyuguang  @wu-sheng need them or not. Thanks.

**Comment by tristaZero** [comment]
@wu-sheng Got it. It is pleasing to see this plugin merged!
@zhaoyuguang Nice job 😉

**Pull request merged**